repositories/transfercheck_enrichment.go

```diff
@@ -2,24 +2,248 @@ package repositories
+func (r *TransferCheckEnrichmentRepository) findIpTypeDichotomy(ip netip.Addr) string {
+	left := 0
+	right := len(r.ipTypeRanges) - 1
+
+	for right >= left {
+		mid := left + (right-left)/2
+
+		ipRange := r.ipTypeRanges[mid].ipRange
+		if ipRange.Contains(ip) {
+			return r.ipTypeRanges[mid].ipType
+		} else if ip.Less(ipRange.Addr()) {
+			right = mid - 1
+		} else {
+			left = mid + 1
+		}
+	}
+
+	return ""
+}
+
+// Expects two CSV files:
+// one with one column (CIDR IP range) containing both ipv4 and ipv6 ranges with VPN address ranges
+// one with one column (IP addresses) containing  both ipv4 and ipv6 with TOR exit nodes
+// The ips & ranges between the two files may overlap.
+func (r *TransferCheckEnrichmentRepository) setupIpTypeRanges(ctx context.Context) error {
+	tracer := utils.OpenTelemetryTracerFromContext(ctx)
+	ctx, span := tracer.Start(
+		ctx,
+		"repositories.TransferCheckEnrichmentRepository.setupIpTypeRanges",
+	)
+	defer span.End()
+	r.muIpTypes.Lock()
+	defer r.muIpTypes.Unlock()
+	file, err := r.gcsRepository.GetFile(ctx, r.bucket, IP_VPN_RANGE_FILE)
+	if err != nil {
+		return errors.Wrap(err, "failed to get VPN IP file")
+	}
+	defer file.Reader.Close()
+	fileReader := csv.NewReader(file.Reader)
+	record, err := fileReader.Read()
+	var ipRange netip.Prefix
+	for err == nil {
+		ipRange, err = netip.ParsePrefix(record[0])
+		if err != nil {
+			return errors.Wrapf(err, "failed to parse VPN IP range %s", record[0])
+		}
+		r.ipTypeRanges = append(r.ipTypeRanges, ipTypeRange{
+			ipRange: ipRange,
+			ipType:  models.VpnIP,
+		})
+		record, err = fileReader.Read()
+	}
+	if err != io.EOF {
+		return errors.Wrap(err, "failed to read VPN IP file")
+	}
+
+	file, err = r.gcsRepository.GetFile(ctx, r.bucket, IP_TOR_RANGE_FILE)
+	if err != nil {
+		return errors.Wrap(err, "failed to get TOR IP file")
+	}
+	defer file.Reader.Close()
+	fileReader = csv.NewReader(file.Reader)
+	record, err = fileReader.Read()
+	var ip netip.Addr
+	for err == nil {
+		ip, err = netip.ParseAddr(record[0])
+		if err != nil {
+			return errors.Wrapf(err, "failed to parse TOR IP address %s", record[0])
+		}
+		r.ipTypeRanges = append(r.ipTypeRanges, ipTypeRange{
+			ipRange: netip.PrefixFrom(ip, ip.BitLen()),
+			ipType:  models.TorIP,
+		})
+		record, err = fileReader.Read()
+	}
+	if err != io.EOF {
+		return errors.Wrap(err, "failed to read TOR IP file")
+	}
+
+	// at the end, sort the full slice of ip ranges
+	slices.SortFunc(r.ipTypeRanges, func(range1, range2 ipTypeRange) int {
+		// multiple ip ranges may have the same start address, but we don't care about how there ordered between them
+		if range1.ipRange == range2.ipRange {
+			return 0
+		} else if range1.ipRange.Addr().Less(range2.ipRange.Addr()) {
+			return -1
+		}
+		return 1
+	})
+
+	return nil
+}
+
+func (r *TransferCheckEnrichmentRepository) GetSenderBicRiskLevel(ctx context.Context, bic string) (string, error) {
```

Comment:
For this one: pending discussion with Arnaud on what to do with it. But we can already go ahead with the current scope of the PR.